akka-http-cluster-management/src/main/scala/akka/cluster/http/management/ClusterHttpManagement.scala

```diff
@@ -0,0 +1,366 @@
+   */
+  def create(cluster: Cluster, https: ConnectionContext): ClusterHttpManagement =
+    apply(cluster, https)
+
+  /**
+   * Creates an instance of [[akka.cluster.http.management.ClusterHttpManagement]] to manage the specified
+   * [[akka.cluster.Cluster]] instance. This version provides Basic Authentication with the specified
+   * AsyncAuthenticator. It does not provide SSL security. It uses the specified path `pathPrefix`.
+   */
+  def create(cluster: Cluster, pathPrefix: String, asyncAuthenticator: AsyncAuthenticator[String]): ClusterHttpManagement =
+    apply(cluster, pathPrefix, asyncAuthenticator)
+
+  /**
+   * Creates an instance of [[akka.cluster.http.management.ClusterHttpManagement]] to manage the specified
+   * [[akka.cluster.Cluster]] instance. This version provides SSL with the specified ConnectionContext.
+   * It does not provide Basic Authentication. It uses the specified path `pathPrefix`.
+   */
+  def create(cluster: Cluster, pathPrefix: String, https: ConnectionContext) =
+    apply(cluster, pathPrefix, https)
+
+  /**
+   * Creates an instance of [[akka.cluster.http.management.ClusterHttpManagement]] to manage the specified
+   * [[akka.cluster.Cluster]] instance. This version provides Basic Authentication with the specified
+   * AsyncAuthenticator. It provide SSL with the specified ConnectionContext. It uses the default path "members".
+   */
+  def create(cluster: Cluster, asyncAuthenticator: AsyncAuthenticator[String], https: ConnectionContext): ClusterHttpManagement =
+    apply(cluster, asyncAuthenticator, https)
+
+  /**
+   * Creates an instance of [[akka.cluster.http.management.ClusterHttpManagement]] to manage the specified
+   * [[akka.cluster.Cluster]] instance. This version provides Basic Authentication with the specified
+   * AsyncAuthenticator. It provide SSL with the specified ConnectionContext. It uses the specified path `pathPrefix`.
+   */
+  def create(cluster: Cluster, pathPrefix: String, asyncAuthenticator: AsyncAuthenticator[String], https: ConnectionContext): ClusterHttpManagement =
+    apply(cluster, pathPrefix, asyncAuthenticator, https)
+}
+
+/**
+ * Class to instantiate an [[akka.cluster.http.management.ClusterHttpManagement]] to
+ * provide an HTTP management interface for [[akka.cluster.Cluster]].
+ */
+class ClusterHttpManagement(
+  cluster:            Cluster,
+  pathPrefix:         Option[String]                     = None,
+  asyncAuthenticator: Option[AsyncAuthenticator[String]] = None,
+  https:              Option[ConnectionContext]          = None) {
+
+  private val settings = new ClusterHttpManagementSettings(cluster.system.settings.config)
+  private implicit val system = cluster.system
+  private implicit val materializer = ActorMaterializer()
+  import system.dispatcher
+
+  private val bindingFuture = new AtomicReference[Future[Http.ServerBinding]]()
+
+  def start(): ClusterHttpManagementServerInfo = {
+    if (bindingFuture.get() != null) {
```

Comment:
use compare and swap of the `AtomicReference` to start it only once, something like:

```scala
val serverBindingPromise = Promise[Http.ServerBinding]()
if (bindingFuture.compareAndSet(null, serverBindingPromise.future)) {
  ...
  serverBindingPromise.completeWith(serverFutureBinding)
}  
```

but I'm not sure it makes sense to support `stop` followed by `start` again. Then things will become complicated, because you can't really start until the `unbind` is completed, and I see no use case for that.